crates/invoker-impl/src/invocation_task/service_protocol_runner.rs

```diff
@@ -220,12 +220,13 @@ where
             DeploymentType::Http {
                 address,
                 protocol_type,
+                http_version,
             } => Endpoint::Http(
                 address,
-                match protocol_type {
+                http_version.unwrap_or(match protocol_type {
                     ProtocolType::RequestResponse => http::Version::default(),
                     ProtocolType::BidiStream => http::Version::HTTP_2,
```

Comment:
Can we encapsulate this logic in a function so that it can be shared with the `render_deployment_type` to avoid duplication?